packages/contentstack-migration/src/utils/modules.js

```diff
@@ -0,0 +1,134 @@
+const fs = require('fs');
+const { execSync } = require('child_process');
+const path = require('path');
+const { sanitizePath } = require('@contentstack/cli-utilities');
+const os = require('os');
+const { builtinModules } = require('module');
+
+const internalModules = new Set(builtinModules);
+
+function checkWritePermissionToDirectory(directory) {
+  try {
+    fs.accessSync(directory, fs.constants.W_OK);
+    return true;
+  } catch (err) {
+    console.log(`You don't have write access to directory`);
+    return false;
+  }
+}
+
+function doesPackageJsonExist(directory) {
+  return fs.existsSync(path.join(sanitizePath(directory), 'package.json'));
+}
+
+function scanDirectory(directory) {
+  return fs.readdirSync(directory);
+}
+
+function scanFileForDependencies(directory, files) {
+  const dependencies = new Set();
+
+  files.forEach((file) => {
+    const filePath = path.join(sanitizePath(directory), sanitizePath(file));
+    if (path.extname(filePath) === '.js') {
+      const fileContent = fs.readFileSync(filePath, 'utf-8');
+      findModulesSync(fileContent).forEach((dep) => dependencies.add(dep));
+    }
+  });
+
+  return [...dependencies];
+}
+
+function createPackageJson(directory) {
+  const templateString = `{
+    "name": "MigrationPackage",
+    "version": "1.0.0",
+    "main": "",
+    "scripts": {},
+    "keywords": [],
+    "author": "",
+    "license": "ISC",
+    "description": ""
+  }`;
+
+  fs.writeFileSync(path.join(sanitizePath(directory), 'package.json'), templateString);
+}
+
+function installDependencies(dependencies, directory) {
```

Comment:
How do we get the version of a dependency ? Installing the latest version of a dependency would be an issue here ??

Reply:
We don't get the versions of the dependency we try to install the latest version for our use case of entry migration that won't be a problem untill we release major version. Asking or letting the user specify specific version will be as good as creating package.json manually or running npm i with specific version. The issue might arise with user running their own script with specific version